MonoGame.Framework/Random.cs

```diff
@@ -0,0 +1,254 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.Xna.Framework
+{
+    public static partial class MathHelper
+    {
+        /// <summary>
+        /// Class for generating random numbers consistantly when using Mono or Microsoft compilers
+        /// </summary>
+        public class Random
+        {
+            #region Static methods
+
+            [ClsCompliant(false)]
+            public static ulong GenerateSeed()
+            {
+                byte[] guid = new Guid().ToByteArray();
+                return (BitConverter.ToUInt64(guid, 0) ^ BitConverter.ToUInt64(guid, 7));
+            }
+
+            #endregion
+
+            #region Private Fields
+
+            private ulong _seed;
+            private ulong _lastValue;
+
+            #endregion
+
+            #region Properties
+
+            [ClsCompliant(false)]
+            public ulong Seed
+            {
+                get
+                {
+                    return this._seed;
+                }
+                set
+                {
+                    this._seed = value;
+                    this._lastValue = value;
```

Comment:
Reading the seed is very important to multiplayer games and sometimes saving game state.

If you need to roll back to a previous game state to replay a missed multiplayer input packet... you have to be able to predictably restore the randoms to the exact state then re-simulate back to the current frame.  So in this case you would store the random seed for each frame to be able to roll back later.

Some games might want to preserve the random in its current state at saving.  With being able to get the seed you can easily restore it and continue that run.  For example maybe a game that is based on generating the starting play state based on a seed itself.